test/BlockManager.js

```diff
@@ -517,6 +517,50 @@ describe('BlockManager', function () {
       assertBNEqual(dispute.lastVisitedStaker, sortedStakers[sortedStakers.length - 1], 'lastVisited should match');
       await blockManager.connect(signers[19]).finalizeDispute(epoch, 0);
     });
+    it('staker should not be able to propose when not elected', async function () {
+      await mineToNextEpoch();
+      const epoch = await getEpoch();
+      const stakerIdAcc8 = await stakeManager.stakerIds(signers[8].address);
+      const staker = await stakeManager.getStaker(stakerIdAcc8);
+      const votes = [100, 200, 300, 400, 500, 600, 700, 800, 900];
+      const commitment1 = utils.solidityKeccak256(
+        ['uint32', 'uint48[]', 'bytes32'],
+        [epoch, votes, '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd']
+      );
+      await voteManager.connect(signers[8]).commit(epoch, commitment1);
+      await mineToNextState();
+      await voteManager.connect(signers[8]).reveal(epoch, votes,
+        '0x727d5c9e6d18ed15ce7ac8d3cce6ec8a0e9c02481415c0823ea49d847ccb9ddd');
+      await mineToNextState();
+      const { biggestInfluencerId } = await getBiggestInfluenceAndId(stakeManager);
+      const iteration = await getIteration(voteManager, stakeManager, staker);
+      const tx = blockManager.connect(signers[8]).propose(epoch,
+        [100, 200, 300, 400, 500, 600, 700, 800, 900],
+        iteration - 1,
```

Comment:
incorrect logic. actually check whether the iteration is incorrect.

Reply:
ok. Actually I wrote this testcase to cover 
require(isElectedProposer(iteration, biggestInfluencerId, proposerId), "not elected");

Reply:
Yeah in that case it won't work

Reply:
I will create another function with a condition